utils/constraint_basics.py

```diff
@@ -0,0 +1,72 @@
+from typing import Union
+
+import hail as hl
+
+from .generic import *
+
+def pre_process_data(ht: hl.Table, split_context_ht_path: str,
+                     output_ht_path: str, overwrite: bool = False) -> None:
+    """ 
+    Add annotations from VEP context Table to gnomAD data.
+    
+    Function adds the following annotations:
+        - context
+        - methylation
+        - coverage
+        - gerp
+        - pass_filters
+    
+    Function drops `a_index`, `was_split`, and`colocated_variants` annotations from gnomAD data.
+    
+    .. note::
+        Function expects that multiallelic variants in the VEP context Table have been split.
+    
+    :param ht: gnomAD exomes or genomes public Hail Table. 
+    :param split_context_ht_path: Path to VEP context Table.
+    :param output_ht_path: Path to output Table.
+    :param overwrite: Whether to overwrite existing data. Defaults to False.
+    """
+    context_ht = hl.read_table(split_context_ht_path).drop('a_index', 'was_split')
+    context_ht = context_ht.annotate(vep=context_ht.vep.drop('colocated_variants'))
+    ht.annotate(**context_ht[ht.key], pass_filters=hl.len(ht.filters) == 0).write(output_ht_path, overwrite)
+
+def prepare_ht(ht: hl.Table, trimer: bool = False, annotate_coverage: bool = True):
```

Comment:
function name could be more descriptive, maybe 'prepare_ht_for_constraint_calculations'